src/hardware/mouse/mouse_config.cpp

```diff
@@ -0,0 +1,288 @@
+// #define ENABLE_EXPLORER_MOUSE
+
+
+MouseConfig     mouse_config;
+MousePredefined mouse_predefined;
+
+static std::vector<std::string> list_models_ps2 = {
+    "standard",
+    "intellimouse",
+#ifdef ENABLE_EXPLORER_MOUSE
+    "explorer",
+#endif
+};
+
+static std::vector<std::string> list_models_com = {
+    "2button",
+    "3button",
+    "wheel",
+    "msm",
+    "2button+msm",
+    "3button+msm",
+    "wheel+msm",
+};
+
+static std::vector<uint16_t> list_rates = {
+// Commented out values are probably not interesting
+// for the end user as "boosted" sampling rate
+//  10",  // PS/2 mouse
+//  20",  // PS/2 mouse
+//  30",  // bus/InPort mouse
+    40,  // PS/2 mouse, approx. limit for 1200 baud serial mouse
+//  50,  // bus/InPort mouse
+    60,  // PS/2 mouse, used by Microsoft Mouse Driver 8.20
+    80,  // PS/2 mouse, approx. limit for 2400 baud serial mouse
+    100, // PS/2 mouse, bus/InPort mouse, used by CuteMouse 2.1b4
+    125, // USB mouse (basic, non-gaming), Bluetooth mouse
+    160, // approx. limit for 4800 baud serial mouse
+    200, // PS/2 mouse, bus/InPort mouse
+    250, // USB mouse (gaming)
+    330, // approx. limit for 9600 baud serial mouse
+    500, // USB mouse (gaming)
+
+// Todays gaming USB mice are capable of even higher sampling
+// rates (like 1000 Hz), but such values are way higher than
+// anything DOS games were designed for; most likely such rates
+// would only result in emulation slowdowns and compatibility
+// issues.
+};
+
+bool MouseConfig::ParseSerialModel(const std::string &model_str,
+                                   MouseModelCOM &model,
+                                   bool &auto_msm)
+{
+    if (model_str == list_models_com[0]) {
+        model    = MouseModelCOM::Microsoft;
+        auto_msm = false;
+        return true;
+    } else if (model_str == list_models_com[1]) {
+        model    = MouseModelCOM::Logitech;
+        auto_msm = false;
+        return true;
+    } else if (model_str == list_models_com[2]) {
+        model    = MouseModelCOM::Wheel;
+        auto_msm = false;
+        return true;
+    } else if (model_str == list_models_com[3]) {
+        model = MouseModelCOM::MouseSystems;
+        auto_msm = false;
+        return true;
+    } else if (model_str == list_models_com[4]) {
+        model    = MouseModelCOM::Microsoft;
+        auto_msm = true;
+        return true;
+    } else if (model_str == list_models_com[5]) {
+        model    = MouseModelCOM::Logitech;
+        auto_msm = true;
+        return true;
+    } else if (model_str == list_models_com[6]) {
+        model    = MouseModelCOM::Wheel;
+        auto_msm = true;
+        return true;
+    }
+
+    return false;
+}
+
+const std::vector<uint16_t> &MouseConfig::GetValidMinRateList()
+{
+    return list_rates;
+}
+
+static void config_read(Section *section)
+{
+    assert(section);
+    const Section_prop *conf = dynamic_cast<Section_prop *>(section);
+    assert(conf);
+    if (!conf)
+        return;
```

Comment:
Is this unneeded with the assert right above?

Reply:
Theoretically `dynamic_cast` can fail if someone attempts a configuration system rework - and this code is not being executed often. I would prefer to leave it as is.